manager/apis/app/v1alpha1/m4dmodule_types.go

```diff
@@ -119,6 +119,25 @@ type Capability struct {
 	Actions []pb.EnforcementAction `json:"actions,omitempty"`
 }
 
+// ExpectedResourceStatus is used to determine the status of an orchestrated resource
+type ExpectedResourceStatus struct {
+	// Kind provides information about the resource kind
+	// +required
+	Kind string `json:"kind"`
+
+	// SuccessCondition specifies a condition that indicates that the resource is ready
+	// +required
```

Comment:
The entire structure is optional, i.e. if a module does not specifies any conditions, we allow it and will use kstatus instead. However, if it does - the readiness indication is a must, otherwise we would never know if the resource is in progress or has completed its task. If a module does not specify a failure condition, we may not know that it has failed but at least we will not wrongly report that the data is ready to be accessed.